cfme/common/__init__.py

```diff
@@ -126,3 +119,50 @@ def validate_properties(self):
             assert(expected_value == shown_value,
                    'Property {} has wrong value, expected {} but was {}'
                    .format(property_tuple[1], expected_value, shown_value))
+
+
+class SummaryPage(object):
+    """
+    Middleware provider and other pages in middleware must extend this class
+    to get table details of `Properties`, `Relationships` and `Smart Management`
+    """
+
+    def properties_ui(self):
+        """
+        `Properties` table
+        Returns: dict value of table details
+        """
+        return self.summary_data_ui("Properties")
+
+    def relationships_ui(self):
+        """
+        `Relationships` table
+        Returns: dict value of table details
+        """
+        return self.summary_data_ui("Relationships")
+
+    def smart_management_ui(self):
+        """
+        `Smart Management` table
+        Returns: dict value of table details
+        """
+        return self.summary_data_ui("Smart Management")
+
+    def summary_data_ui(self, header):
```

Comment:
@jkrocil Thank you! I am sorry I didn't notice `self.get_detail`. I will replace as per your recommendation.
* To get multiple values under a tables I am following for loop, for example I want to get Properties information from middleware provider summary page. Properties table has `Name`, `Type`, `Host Name` and `Port` values. Page is navigating 4 times from provider page to summary page.
```
pro_fn = functools.partial(self.get_detail, "Properties")
data = {}
for arg in ['Name', 'Type', 'Host Name', 'Port']:
  data.update({arg:pro_fn(arg)})
return data
```
Is there any option to get all these values without navigation? When we go to deployment page, navigation will happen like, `Providers >> Provider Summary >> Servers >> Server Summary >> Deployments >> Deployment Summary`
* In some places, pages running without provider. Is there any way to call `get_detail` function without provider object?